From 451c8e45200aeb67c56d42e5398e7c12466b30d6 Mon Sep 17 00:00:00 2001
From: Adrian Reber <adrian@lisas.de>
Date: Fri, 2 Dec 2016 12:16:50 +0100
Subject: [PATCH 1/3] configure.ac: adapt to openssl-1.1

openssl-1.1 removed the function CRYPTO_lock() which was used
in the configure script to check for openssl availability.
This changes CRYPTO_lock() to HMAC_Update().

Fixes: #125

Signed-off-by: Adrian Reber <adrian@lisas.de>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index c8a933c..d38e290 100644
--- a/configure.ac
+++ b/configure.ac
@@ -406,9 +406,9 @@ if test "x-$want_ssl" != "x-no" ; then
     fi
     AC_CHECK_HEADERS(openssl/crypto.h)
     if test "x-$ac_cv_header_openssl_crypto_h" = "x-yes" ; then
-        AC_CHECK_LIB(crypto, CRYPTO_lock)
+        AC_CHECK_LIB(crypto, HMAC_Update)
     fi
-    if test "x-$ac_cv_lib_crypto_CRYPTO_lock" = "x-yes" ; then
+    if test "x-$ac_cv_lib_crypto_HMAC_Update" = "x-yes" ; then
         AC_CHECK_HEADERS(openssl/ssl.h)
     fi
     if test "x-$ac_cv_header_openssl_ssl_h" = "x-yes" ; then
-- 
2.9.3

